### PR TITLE
ci: check public headers conform to c++ ISO standards

### PR DESCRIPTION
**Pull request by 64**
Previously only C ISO versions were checked. Check C++ versions too.

### CODE REVIEW

**Review by no92 — COMMENTED** [review]
Apart from my comments, this LGTM.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -64,12 +64,17 @@ jobs:
                     ;;
                 esac
                 ALL_HDRS="packages/${{matrix.builds}}/usr/include/**.h"
-                GCC_ARGS="$GCC_ARCH-linux-mlibc-gcc -x c -S /dev/null -o /dev/null -I packages/${{matrix.builds}}/usr/include -I packages/linux-headers/usr/include -nostdlib -Werror -Wpedantic -Wsystem-headers"
+                GCC_ARGS="-S /dev/null -o /dev/null -I packages/${{matrix.builds}}/usr/include -I packages/linux-headers/usr/include -nostdlib -Werror -Wpedantic -Wsystem-headers"
```
[the remark this answers is not in the document]
Comment (64) [reply .github/workflows/ci.yml:67]: I guess so, but they're clean?

[the remark this answers is not in the document]
Comment (64) [reply .github/workflows/ci.yml:67]: At least the bits we use in mlibc like syscall numbers

**Review by Dennisbonke — APPROVED** [review]
LGTM

**Pull request merged**